types/src/v0/impls/state.rs

```diff
@@ -694,6 +695,38 @@ fn validate_builder_fee(
     Ok(())
 }
 
+async fn update_l1_state_stake_tables(
+    instance: &NodeState,
+    header: &Header,
+    chain_config: ChainConfig,
+) -> anyhow::Result<()> {
+    let Some(contract_addr) = chain_config.stake_table_contract else {
+        bail!("stake table contract not found");
+    };
+
+    let prev_epoch = epoch_from_block_number(header.block_number().saturating_sub(1), EPOCH_HEIGHT);
+    let epoch = epoch_from_block_number(header.block_number(), EPOCH_HEIGHT);
+
+    if epoch <= prev_epoch {
+        return Ok(());
```

Comment:
Should this be an error... Or, if we don't want this to propagate up to `BlockError` we should at least have an `info!` or `warn!`.

Reply:
5685090edc97c99366c45b6fd45b762dc82d5e77